### PR TITLE
feat(chart): implement `apiObjects` function

### PR DESCRIPTION
**Pull request by shinebayar-g**
Fixes #

This PR implements `get apiObjects(): ApiObject[]` function at the chart level. So that users can easily access all the k8s objects of the chart. Borrowed from [Include class.](https://github.com/cdk8s-team/cdk8s-core/blob/8da5c95703f438f375fe003d33966a06c45ce7b7/src/include.ts#L37-L42)

### CODE REVIEW

**Comment by shinebayar-g** [comment]
Maybe I should add a test case like this https://github.com/cdk8s-team/cdk8s-core/blob/c302999893aa3dffc9754c3f5d129135d41bbcb4/test/include.test.ts#L61C1-L82C6

**Comment by vinayak-kukreja** [comment]
Hey, could you please tell me your use case for adding this functionality?

**Comment by shinebayar-g** [comment]
Our use case is to work with the `Chart` instance the same way we use `Include`. If Chart instance also has this function it enables us to do more things like this https://github.com/cdk8s-team/cdk8s-plus/issues/4239

**Comment by heckler1** [comment]
Just as an observer, this piqued my interest so I was doing some testing locally. Is there a difference between just running this transformation in your own cdk8s code vs adding it to the library?

I can see that 
```
    myChart.node.children
      .filter((o): o is ApiObject => o instanceof ApiObject)
      .forEach((apiObject) => {
          apiObject.addJsonPatch(
            JsonPatch.replace("/metadata/namespace", "mynamespace"),
          );
      });
```
works just fine on my local repo. Is this just adding a convenience function?

**Comment by shinebayar-g** [comment]
Yeah it's just a convenience function that exists on Include class, porting it over to Chart class to make the API consistent.

**Comment by vinayak-kukreja** [comment]
> Our use case is to work with the `Chart` instance the same way we use `Include`. If Chart instance also has this function it enables us to do more things like this [cdk8s-team/cdk8s-plus#4239 (comment)](https://github.com/cdk8s-team/cdk8s-plus/issues/4239)

Hey @shinebayar-g, this makes sense to me. Thank you for making this contribution. 
Could you add a test for this functionality?

**Comment by shinebayar-g** [comment]
Hey @vinayak-kukreja , I added a test case. Pls let me know if it works.
When I run `yarn test`, helm test case is failing.

```
 PASS  test/metadata.test.ts (8.464 s)
 PASS  test/size.test.ts (8.727 s)
 PASS  test/names.test.ts (8.804 s)
 PASS  test/names-legacy.test.ts (8.724 s)
 PASS  test/app.test.ts (8.826 s)
 PASS  test/api-object.test.ts (8.952 s)
 PASS  test/yaml.test.ts (9.39 s)
 PASS  test/duration.test.ts
 PASS  test/dependency.test.ts
 PASS  test/util.test.ts
 PASS  test/cron.test.ts
 PASS  test/include.test.ts (9.429 s)
 PASS  test/json-patch.test.ts
 PASS  test/tokens.test.ts
 PASS  test/chart.test.ts
 FAIL  test/helm.test.ts (11.503 s)
  ● fails if the helm executable is not found

    expect(received).toThrow(expected)

    Expected substring: "unable to execute 'foo-347951.4027040863' to render Helm chart"
    Received message:   "error while rendering a helm chart: spawnSync foo-347951.4027040863 EACCES"

          146 |     }
          147 |
        > 148 |     throw new Error(`error while rendering a helm chart: ${err}`);
              |           ^
          149 |   }
          150 |
          151 |   if (helm.status !== 0) {

          at renderTemplate (src/helm.ts:148:11)
          at new Helm (src/helm.ts:129:24)
          at test/helm.test.ts:29:16
          at Object.<anonymous> (node_modules/expect/build/toThrowMatchers.js:83:11)
          at Object.throwingMatcher [as toThrow] (node_modules/expect/build/index.js:382:21)
          at Object.<anonymous> (test/helm.test.ts:32:7)

      30 |     chart: SAMPLE_CHART_PATH,
      31 |     helmExecutable: executable,
    > 32 |   })).toThrow(`unable to execute '${executable}' to render Helm chart`);
         |       ^
      33 | });
      34 |
      35 | test('values can be specified when defining the chart', () => {

      at Object.<anonymous> (test/helm.test.ts:32:7)

--------------------|---------|----------|---------|---------|-----------------------------------------
File                | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
--------------------|---------|----------|---------|---------|-----------------------------------------
All files           |   98.81 |    94.66 |   95.36 |   98.81 |
 src                |    98.8 |    94.63 |   95.33 |    98.8 |
  _child_process.ts |     100 |      100 |     100 |     100 |
  _util.ts          |     100 |      100 |     100 |     100 |
  api-object.ts     |   98.76 |    96.66 |     100 |   98.76 | 241-243
  app.ts            |   98.82 |    96.15 |     100 |   98.82 | 254,303-304,308-309
  chart.ts          |     100 |      100 |     100 |     100 |
  cron.ts           |     100 |      100 |     100 |     100 |
  dependency.ts     |     100 |      100 |     100 |     100 |
  duration.ts       |   95.78 |    89.58 |   90.47 |   95.78 | 69-70,73-74,133-134,159,201-202,250-251
  helm.ts           |   98.74 |    93.75 |     100 |   98.74 | 145-146
  include.ts        |     100 |    55.55 |     100 |     100 | 29-32
  index.ts          |     100 |      100 |     100 |     100 |
  json-patch.ts     |     100 |      100 |     100 |     100 |
  lazy.ts           |     100 |      100 |     100 |     100 |
  metadata.ts       |   99.33 |    85.71 |   88.88 |   99.33 | 196-197
  names.ts          |   98.08 |    93.33 |   88.88 |   98.08 | 134-135,160-161
  resolve.ts        |     100 |      100 |     100 |     100 |
  size.ts           |     100 |    96.15 |     100 |     100 | 149
  testing.ts        |   95.34 |    83.33 |      75 |   95.34 | 41-42
  yaml.ts           |   96.39 |    94.11 |   71.42 |   96.39 | 22-23,97-98
 test               |     100 |      100 |     100 |     100 |
  util.ts           |     100 |      100 |     100 |     100 |
--------------------|---------|----------|---------|---------|-----------------------------------------
Test Suites: 1 failed, 15 passed, 16 total
Tests:       1 failed, 181 passed, 182 total
Snapshots:   42 passed, 42 total
Time:        12.601 s
Ran all test suites.
👾 Task "test" failed when executing "jest --passWithNoTests --coverageProvider=v8 --updateSnapshot" (cwd: /home/user/github/cdk8s-core)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I have `helm` installed and it's on `PATH`.

Also my prettier formatter picked up some changes. I think they're overall looks good. This is the prettier configuration that I used. If you feel this is unnecessary I can revert.

```json
{
  "tabWidth": 2,
  "singleQuote": true,
  "printWidth": 100,
  "trailingComma": "all"
}
```

**Comment by vinayak-kukreja** [comment]
Hey @shinebayar-g , thank you for adding the tests. I will try to run this locally. 
Apologies for the delay, I will try to get back to you asap.

**Comment by vinayak-kukreja** [comment]
Hey @shinebayar-g , apologies for the delay 🙏 

Could you please remove the changes being added by prettier?

**Comment by shinebayar-g** [comment]
Okay, a sec.

**Comment by shinebayar-g** [comment]
Done. Btw `yarn test` is passing for me this time. `test/helm.test.ts` is no longer failing :)

File: test/chart.test.ts [hunk 1]
```diff
@@ -142,7 +142,6 @@ test('synthesizeManifest() can be used to synthesize a specific chart', () => {
 });
 
 test('addDependency', () => {
-
```
Comment (sumupitchayan) [inline test/chart.test.ts]: @shinebayar-g can you also remove the change of these deleted lines? Think this was also caused by prettier formatting

**Review by sumupitchayan — APPROVED** [review]

**Comment by sumupitchayan** [comment]
## 💔 Backport failed
The pull request could not be backported due to the following error:
`Missing target branch choices`

### Manual backport
To create the backport manually run:
```
backport --pr 1795
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by sumupitchayan** [comment]
## 💔 Backport failed
The pull request could not be backported due to the following error:
`Missing target branch choices`

### Manual backport
To create the backport manually run:
```
backport --pr 1795
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by sumupitchayan** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|1.x|[](https://github.com/cdk8s-team/cdk8s-core/pull/1902)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**